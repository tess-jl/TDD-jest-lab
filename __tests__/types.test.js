const {
  isNumber,
  isString,
  castToNumber,
  getCaster
} = require('../lib/types.js');

describe('validator module', () => {
  describe('basic validation', () => {
    it('properly tells if a value is a number', () => {
      expect(isNumber(3)).toBeTruthy();
      expect(isNumber('hi')).toBeFalsy();
      expect(isNumber([])).toBeFalsy();
      expect(isNumber({})).toBeFalsy();
      expect(isNumber(() => {})).toBeFalsy();
      expect(isNumber(true)).toBeFalsy();
    });
    it('properly tells if a value is a string', () => {
      expect(isString('hi')).toBeTruthy();
      expect(isString('')).toBeTruthy(); 
      expect(isString(2)).toBeFalsy();
      expect(isString([])).toBeFalsy();
      expect(isString({})).toBeFalsy();
      expect(isString(() => {})).toBeFalsy();
      expect(isString(true)).toBeFalsy();
    });


  });

  describe('casters', () => {
    it('can cast values to a number', () => {
      expect(castToNumber(3)).toEqual(3);
      expect(castToNumber('3')).toEqual(3);
      expect(castToNumber(true)).toEqual(1);
      expect(castToNumber(false)).toEqual(0);
    });

    it('throws if value is not castable to number', () => {
      expect(() => castToNumber('hi')).toThrowErrorMatchingSnapshot();
      expect(() => castToNumber({})).toThrowErrorMatchingSnapshot();
    });

    // it('can cast values to a string', () => {
    //   expect(castToString(3)).toEqual('3');
    //   expect(castToString('3')).toEqual(3);
    //   expect(castToString(true)).toEqual(1);
    //   expect(castToString(false)).toEqual(0);
    // });
  });

  it('can get the right caster', () => {
    expect(getCaster(Number)).toEqual(castToNumber);
    expect(getCaster(Promise)).toBeNull();
  });
});
