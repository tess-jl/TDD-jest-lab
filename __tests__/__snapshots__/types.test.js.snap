// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator module casters throws error if value is not castable to number 1`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws error if value is not castable to number 2`] = `"Cannot cast >>[object Object]<< to Number"`;

exports[`validator module casters throws error if value is not castable to number 3`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws error if value is not castable to number 4`] = `"Cannot cast >>[object Object]<< to Number"`;

exports[`validator module casters throws error if value is not castable to string 1`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws error if value is not castable to string 2`] = `"Cannot cast >>[object Object]<< to Number"`;

exports[`validator module casters throws if value is not castable to number 1`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws if value is not castable to number 2`] = `"Cannot cast >>[object Object]<< to Number"`;

exports[`validator module casters throws if value is not castable to number 3`] = `"Cannot cast >>hi<< to Number"`;

exports[`validator module casters throws if value is not castable to number 4`] = `"Cannot cast >>[object Object]<< to Number"`;
